partiql-tests-data/eval/primitives/path.ion

```diff
@@ -353,412 +433,480 @@ path::[
+    statement: "test_struct.\"unambiguous_field\"",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:"this is unambiguous_field"
+      output: "this is unambiguous_field"
     }
   },
   {
-    name:"dotted path expression with quoted field name accesses field UNAMBIGUOUS_FIELD (uppercase)",
-    statement:"test_struct.\"UNAMBIGUOUS_FIELD\"",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
-        EvalModeCoerce,
-        EvalModeError
-      ],
-      output:$missing::null
-    }
+    name: "dotted path expression with quoted field name accesses field UNAMBIGUOUS_FIELD (uppercase)",
+    statement: "test_struct.\"UNAMBIGUOUS_FIELD\"",
+    assert: [
+      {
+        result: EvaluationSuccess,
+        evalMode: EvalModeCoerce,
+        output: $missing::null
+      },
+      {
+        evalMode: EvalModeError,
+        result: EvaluationFail
+      }
+    ]
   },
   {
-    name:"dotted path expression with quoted field name accesses field ambiguous_field (lowercase)",
-    statement:"test_struct.\"ambiguous_field\"",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "dotted path expression with quoted field name accesses field ambiguous_field (lowercase)",
+    statement: "test_struct.\"ambiguous_field\"",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:"this is ambiguous_field (lowercase)"
+      output: "this is ambiguous_field (lowercase)"
     }
   },
   {
-    name:"dotted path expression with quoted field name accesses field AMBIGUOUS_FIELD (uppercase)",
-    statement:"test_struct.\"AMBIGUOUS_FIELD\"",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "dotted path expression with quoted field name accesses field AMBIGUOUS_FIELD (uppercase)",
+    statement: "test_struct.\"AMBIGUOUS_FIELD\"",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:"this is AMBIGUOUS_FIELD (uppercase)"
+      output: "this is AMBIGUOUS_FIELD (uppercase)"
     }
   },
+  // This has the below behavior due to PartiQL Specification: "The expression a[s] is a shorthand for the tuple path navigation a.s when
+  //  the expression s is either (a) a string literal or (b) an expression that is explicitly CAST into a string."
   {
-    name:"subscript with variable in lowercase",
-    statement:"test_struct[unambiguous_variable]",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
-        EvalModeCoerce,
-        EvalModeError
-      ],
-      output:"this is unambiguous_lookup_field"
-    }
+    name: "subscript with variable in lowercase",
+    statement: "test_struct[unambiguous_variable]",
+    assert: [
+      {
+        result: EvaluationSuccess,
+        evalMode: EvalModeCoerce,
+        output: $missing::null
```

Comment:
Tuple navigation with array notation The expression a[s] is a shorthand for the tuple path navigation a.s when
the expression s is either (a) a string literal or (b) an expression that is explicitly CAST into a string. For example:
{'a': 1, 'b': 2}['a'] ⇔ {'a': 1, 'b':2}.'a' → 1

Not sure what the single quoted 'a' means in our spec. but curious why you think this should return a MISSING...

Reply:
> when the expression s is either (a) a string literal or (b) an expression that is explicitly CAST into a string... If s is not a string literal or an expression that is cast into a string, then a[s] is evaluated as an array path navigation.

So, `some_struct['a']` (literal string `'a'`) results in a path expression. Similarly, `some_struct[CAST('a' || 'b' AS STRING)]` (explicit cast) results in a path expression.

"If s is not a string literal or an expression that is cast into a string" (like variable references, plus operations, etc), "then a[s] is evaluated as an array path navigation." All of the below evaluate as an array path navigation:
1. `some_struct[some_variable]`
2. `some_struct[1 + 1]`
3. `some_struct['hello' || ', world']`

Reply:
Gotya. So `test_struct['unambiguous_variable']` => 'this is unambiguous_lookup_field'`. And the test case is missing single quote here. Is this understanding correct?

Reply:
Well, I think the intention of the test is a bit different. `unambiguous_variable` is a global variable pointing to some string, and I believe the original intention was to test that the underlying value that the variable points to can be used to key into structs -- though that assumption is wrong according to the specification.